app/controllers/comments/votes_controller.rb

```diff
@@ -0,0 +1,35 @@
+module Comments
+  class VotesController < ApplicationController
+    load_and_authorize_resource :comment
+    load_and_authorize_resource through: :comment, through_association: :votes_for, only: :destroy
+    before_action :authenticate_user!
```

Comment:
Definitely strange to authorize the resource before authenticating the user :thinking:.

I think this works by coincidence. We aren't displaying the buttons to destroy a vote when users are signed in, and we aren't using `load_and_authorize_resource` for the create action.

To keep everyone's sanity :exploding_head:, I'd suggest to authenticate users before authorizing the resource, and writing a test that fails like this but passes after the change. For instance:

```ruby
    it "redirects unidentified users to the sign in page" do
      delete :destroy, params: { comment_id: comment.id, id: vote }

      expect(response).to redirect_to new_user_session_path
    end
```